Makefile

```diff
@@ -22,22 +25,40 @@ doccheck:
 	@cd Documentation; for doc in *.md; do if [ "$$doc" != "README.md" ] && ! grep -q "$$doc" *.md; then echo "ERROR: No link to documentation file $${doc} detected"; exit 1; fi; done
 	@echo OK
 
-deps:
-	go get github.com/tools/godep
+build: clean
+	GOOS=$(shell uname -s | tr A-Z a-z) GOARCH=$(ARCH) $(BUILDENVVAR) go build -o kube-state-metrics
+
+test-unit: clean build
+	GOOS=$(shell uname -s | tr A-Z a-z) GOARCH=$(ARCH) $(TESTENVVAR) go test --race $(FLAGS) $(PKGS)
+
+TEMP_DIR := $(shell mktemp -d)
+
+all: all-container
+
+sub-container-%:
+	$(MAKE) ARCH=$* container
+
+sub-push-%:
+	$(MAKE) ARCH=$* push
 
-build: clean deps
-	$(COMMONENVVAR) $(BUILDENVVAR) godep go build -o kube-state-metrics 
+all-container: $(addprefix sub-container-,$(ALL_ARCH))
 
-test-unit: clean deps build
-	$(COMMONENVVAR) $(TESTENVVAR) godep go test --race $(FLAGS) $(PKGS)
+all-push: $(addprefix sub-push-,$(ALL_ARCH))
 
-container: build
-	docker build -t ${REGISTRY}/kube-state-metrics:$(TAG) .
+container: .container-$(ARCH)
+.container-$(ARCH):
+	cp -r * $(TEMP_DIR)
+	GOOS=$(shell uname -s | tr A-Z a-z) GOARCH=$(ARCH) $(BUILDENVVAR) go build -o $(TEMP_DIR)/kube-state-metrics
+	docker build -t $(MULTI_ARCH_IMG):$(TAG) $(TEMP_DIR)
```

Comment:
@Pensu Should add  a check for amd64 arch and build `$(IMAGE):$(TAG)` image or otherwise an error will be fired when later pushing.

Reply:
As mention by @andyxning, we need to add a fallback case for amd64 to tag the image to older format without any arch in it.

Reply:
Have added a check for amd64 arch.